packages/react-components/react-spinbutton/src/components/SpinButton/useSpinButton.tsx

```diff
@@ -163,9 +166,15 @@ export const useSpinButton_unstable = (props: SpinButtonProps, ref: React.Ref<HT
   };
 
   const stepValue = (e: SpinButtonChangeEvent, direction: 'up' | 'down' | 'upPage' | 'downPage') => {
+    const val = internalState.current.value;
     const dir = direction === 'up' || direction === 'upPage' ? 1 : -1;
     const stepSize = direction === 'upPage' || direction === 'downPage' ? stepPage : step;
-    const val = internalState.current.value;
+
+    if (val === null) {
+      const nullStep = clamp(0 + stepSize * dir, min, max);
```

Comment:
Should 0 be hardcoded here, or should we default to starting from the min?

Reply:
The `0` can obviously be removed but I felt leaving makes it clearer where the step is starting from and leaving it has no material impact on performance.

`min` and `max` aren't necessarily both defined. You can define neither, both or only one and `clamp()` takes this into account. The only constraints are that they are both numbers and `min <= max`. `max` can be less than 0 so it's possible the value would be clamped to `max` rather than `min`.

I'll update tests to reflect this.